src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health=health;
+        this.strength=strength;
+    }
+    attack(){
+        return this.strength;
+    }
+    receiveDamage(damage){
+        this.health-=damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage){
+        this.health-=damage;
+
+        if (this.health > 0 ) {return `${this.name} has received ${damage} points of damage`; }
+        else { return `${this.name} has died in act of combat`; }
+    }
+
+    battleCry(){
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage){
+        this.health-=damage;
+        let result;
+        if (this.health >0) {result=`A Saxon has received ${damage} points of damage`; }
+        else { result=`A Saxon has died in combat`; }
+        return result;
+    }
+}
 
 // War
-class War {}
+class War {
+    vikingArmy = [];
+    saxonArmy = [];
+
+    addViking(v) {
+        this.vikingArmy.push(v);
+    }
+    addSaxon(s){
+        this.saxonArmy.push(s);
+    }
+    vikingAttack(){
+        // console.log(`vikingAttack 1`);
+        let vRandom=Math.floor(Math.random() * this.vikingArmy.length);
+        let sRandom=Math.floor(Math.random() * this.saxonArmy.length);
+        const viking=this.vikingArmy[vRandom];
+        const saxon=this.saxonArmy[sRandom];
+        let result=saxon.receiveDamage(viking.strength);
+        if (saxon.health<=0){
+            // console.log(`vikingAttack IN`);
+            this.saxonArmy.pop(sRandom);
```

Comment:
this doesn't work - like we talked about in class. pop doesn't accept an argument afaik, so it will just drop the last element and not do what you think it would - namely pop out the element at position sRandom. Use splice, slice, filter or a for loop instead :+1: